components/service/pocket/src/main/java/mozilla/components/service/pocket/PocketEndpoint.kt

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.pocket
+
+import android.support.annotation.WorkerThread
+import mozilla.components.concept.fetch.Client
+import mozilla.components.service.pocket.data.PocketGlobalVideoRecommendation
+import mozilla.components.service.pocket.net.PocketResponse
+
+// See @Ignore tests in test class to do end-to-end testing.
+
+/**
+ * Makes requests to the Pocket API and returns the requested data.
+ *
+ * @see [getInstance] to retrieve an instance.
+ */
+class PocketEndpoint internal constructor(
+    private val rawEndpoint: PocketEndpointRaw,
+    private val jsonParser: PocketJSONParser
+) {
+
+    /**
+     * Gets a response, filled with the Pocket global video recommendations from the Pocket API server on success.
+     * This network call is synchronous and the results are not cached. The API version of this call is available in the
+     * source code at [PocketURLs.VALUE_VIDEO_RECS_VERSION].
+     *
+     * If the API returns unexpectedly formatted results, these entries will be omitted and the rest of the items are
+     * returned.
+     *
+     * @return a [PocketResponse.Success] with the Pocket global video recommendations (the list will never be empty)
+     * or, on error, a [PocketResponse.Failure].
+     */
+    @WorkerThread // Synchronous network call.
+    fun getGlobalVideoRecommendations(): PocketResponse<List<PocketGlobalVideoRecommendation>> {
+        val json = rawEndpoint.getGlobalVideoRecommendations()
+        val videoRecs = json?.let { jsonParser.jsonToGlobalVideoRecommendations(it) }
+        return videoRecs.wrapInResponse()
+    }
+
+    private fun <T> List<T>?.wrapInResponse(): PocketResponse<List<T>> = if (isNullOrEmpty()) {
+        PocketResponse.Failure()
+    } else {
+        PocketResponse.Success(this!!)
+    }
+
+    companion object {
+
+        /**
+         * Returns an instance of [PocketEndpoint].
+         *
+         * @param client the HTTP client to use for network requests.
+         * @param pocketApiKey the API key for Pocket network requests.
+         * @param userAgent the user agent for network requests.
+         *
+         * @throws IllegalArgumentException if the provided API key or user agent is deemed invalid.
+         */
+        fun getInstance(client: Client, pocketApiKey: String, userAgent: String): PocketEndpoint {
```

Comment:
naming nit: `getInstance()` sounds like the method of a singleton that always returns the same instance. How about. `create()` or `build()` or `createInstance()`?

Reply:
It's possible this getter method actually caches instances under the hood (though I seriously doubt we'll ever do that) so I didn't want to use `create*` or `build` but I can see the problem with `getInstance`. How about, `get`?

Reply:
`get` is a bit better. But I think I would still expect a Singleton instance from it. :)

> (though I seriously doubt we'll ever do that)

Why choose a name that will work with an unlikely future change though? :)

Anyhow, I'm fine with whatever you decide to pick. :)

Reply:
Effective Java seems to imply `getInstance` with parameters doesn't imply singleton: 

> getInstance — Returns an instance that is described by the parameters but cannot be said to have the same value. In the case of a singleton, getInstance takes no parameters and returns the sole instance.

via https://stackoverflow.com/a/22442058

Reply:
So I'm going to leave it as is.

Reply:
> Why choose a name that will work with an unlikely future change though? :)

To be fair, picking a more specific name like `newInstance` or `build` enforces constraints – this will always return a new instance – that maybe we don't want to be confined to.

Reply:
Then again, people's code might assume it returns an newInstance each time so perhaps it's better to name it what it is and create a new method if the behavior changes. I'll do that.